src/schemathesis/extra/pytest_plugin.py

```diff
@@ -278,10 +284,13 @@ def pytest_pyfunc_call(pyfuncitem):  # type:ignore
 
     __tracebackhide__ = True
     if isinstance(pyfuncitem, SchemathesisFunction):
-        with skip_unnecessary_hypothesis_output():
-            outcome = yield
         try:
-            outcome.get_result()
+            with skip_unnecessary_hypothesis_output():
+                if IS_PYTEST_ABOVE_54:
+                    yield
+                else:
+                    outcome = yield
+                    outcome.get_result()
```

Comment:
both the test for the pytest version and the handling of the exceptions below is incorrect for old style
also the version check is wrong for the pytest version and the pluggy version

Reply:
Could you, please, elaborate on the exception handling part? My assumption is that it used to work correctly (based on the tests I have) - if you could highlight what aspect is incorrect, it would be a great help.

Re: version check. You mean that it should check for the `pluggy` version instead of `pytest` right?

Reply:
old hookwrappers simply ignored the exceptions
thats why the tests would possibly pass
pluggy now warns about those exceptions (and i git them in testsuites)
hookwrappers as per the old style must not raise a exception

https://pluggy.readthedocs.io/en/stable/#old-style-wrappers explains the details

but bascially its only valid to use result.force_exception, its not valid to raise a exception

its strongly recommend to only use the new api

or if you want to support unreasonably outdated deployments, correctly implement the hookwrapper

its absolutely incorrect to version-gate the hook specification, pick a style for implementation and then use it

its strongly recommended to use the new style